lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py

```diff
@@ -266,20 +267,15 @@ def process_body_from_alb_invocation(event):
 
 @lru_cache(maxsize=None)
 def read_from_ssm(key: str) -> str:
+    ssm_client = boto3.client('ssm')
+    logger.info('Getting key: %s from SSM', key)
     try:
-        ssm_client = boto3.client('ssm')
-
-        logger.info('Generated ssm_client')
-
         response = ssm_client.get_parameter(Name=key, WithDecryption=True)
-
         logger.info('received ssm parameter')
-
-        return response.get('Parameter', {}).get('Value', '')
     except Exception as e:
-        logger.error('General Exception With Call to VeText')
-        logger.exception(e)
-        return ''
+        logger.critical('Exception raised while looking up SSM key %s. Exception: %s', key, e)
+
+    return response.get('Parameter', {}).get('Value')
```

Comment:
If "get_parameter" raises an exception, "response" is uninitialized, and calling "get" will raise AttributeError.  I would set `response = {}` in the "except" block.

Reply:
Great callout. I'm adding tests for this now.